lib/rules/forbid-component-props.js

```diff
@@ -25,31 +28,68 @@ module.exports = {
     schema: [{
       type: 'object',
       properties: {
+        checkDomNodes: {
+          type: 'boolean'
+        },
         forbid: {
           type: 'array',
           items: {
             type: 'string'
           }
+        },
+        forbidPatterns: {
+          type: 'array',
+          items: {
+            type: 'string'
+          }
         }
       },
       additionalProperties: true
     }]
   },
 
   create: function(context) {
+    var configuration = context.options[0] || {};
+    var checkDomNodes = configuration.checkDomNodes || false;
+    var ignoreComponents = configuration.ignoreComponents || false;
 
-    function isForbidden(prop) {
-      var configuration = context.options[0] || {};
+    function somePatternMatches(patterns, subject) {
+      return patterns.reduce(function (acc, pattern) {
+        if (sanitizeUserPattern(pattern).test(subject)) {
+          return true;
+        }
+        return acc;
+      }, false);
+    }
 
+    function isForbidden(prop) {
       var forbid = configuration.forbid || DEFAULTS;
-      return forbid.indexOf(prop) >= 0;
+      var forbidPatterns = configuration.forbidPatterns;
+      if (forbid.indexOf(prop) >= 0) {
+        return true;
+      }
+
+      if (forbidPatterns) {
+        if (somePatternMatches(forbidPatterns, prop)) {
+          return true;
+        }
+      }
+
+      return false;
     }
 
     return {
       JSXAttribute: function(node) {
         var tag = node.parent.name.name;
-        if (tag && tag[0] !== tag[0].toUpperCase()) {
-          // This is a DOM node, not a Component, so exit.
+        var isComponent = tag && tag[0] === tag[0].toUpperCase();
```

Comment:
i think your code here is making the false assumption that there's only two kinds of nodes, DOM and Component. nodes can also be strings, numbers, arrays… are we sure this is the right way to check "is dom node" and "is component"? I feel like we have functions for this in the codebase already.